pkg/apis/monitoring/v1beta1/alertmanager_config_types.go

```diff
@@ -1291,6 +1295,156 @@ type RocketChatActionConfig struct {
 	Msg *string `json:"msg,omitempty"`
 }
 
+// MattermostConfig defines actions for Mattermost messages.
+type MattermostConfig struct {
+	// sendResolved defines whether or not to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// webhookURL defines the Mattermost webhook URL.
+	// +required
+	WebhookURL *v1.SecretKeySelector `json:"webhookURL,omitempty"`
+	// channel overrides the channel the message posts in.
+	// Use the channel’s name and not the display name, e.g. use town-square, not Town Square.
+	// +kubebuilder:validation:MinLength=1
+	// +optional
+	Channel *string `json:"channel,omitempty"`
+	// username overrides the username the message posts as.
+	// Defaults to the username set during webhook creation; if no username was set during creation, webhook is used.
+	// +optional
+	Username *string `json:"username,omitempty"`
+	// text defines the markdown-formatted message to display in the post.
+	// To trigger notifications, use @<username>, @channel, and @here like you would in other Mattermost messages.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Text string `json:"text"`
```

Comment:
the only required field is WebhookURL - text is only required for the MattermostField type